src/utils/filterExpanses.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const filterExpanses = (expense, queryParams) => {
+  const { userId, categories, from, to } = queryParams;
+
+  return (
+    (!userId || expense.userId === Number(userId))
+    && (!categories || categories.includes(expense.category))
+    && (!from || expense.spentAt >= from)
+    && (!to || expense.spentAt <= to)
```

Comment:
Ensure that the 'from' and 'to' variables are converted to proper date objects before making comparisons.